credit_limit_sale_order_approval/models/res_partner.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+################################################################
+#    License, author and contributors information in:          #
+#    __openerp__.py file at the root folder of this module.    #
+################################################################
+
+from openerp import models, fields, api
+import logging
+_logger = logging.getLogger(__name__)
+
+
+# RES.PARTNER
+class res_partner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.multi
+    @api.depends('credit', 'credit_limit')
+    def _get_available_credit(self):
+        for partner in self:
+            available_credit = partner.credit_limit - partner.credit
+            if partner.set_credit_limit and available_credit > 0:
+                partner.available_credit = available_credit
+            else:
+                partner.available_credit = 0
```

Comment:
All computed fields are reset at the beginning of the calculation to their default, so there's no need of putting this line.